src/query/vacuum.c

```diff
@@ -1092,6 +1092,63 @@ xvacuum (THREAD_ENTRY * thread_p)
 #endif /* SA_MODE */
 }
 
+/*
+ * xvacuum_dump - Dump the contents of vacuum
+ *
+ * return: nothing
+ *
+ *   outfp(in): FILE stream where to dump the vacuum. If NULL is given,
+ *            it is dumped to stdout.
+ */
+void
+xvacuum_dump (THREAD_ENTRY * thread_p, FILE * outfp)
+{
+  LOG_PAGEID min_log_pageid = NULL_PAGEID;
+  int archive_number;
+
+  if (outfp == NULL)
+    {
+      outfp = stdout;
+    }
+
+  /* vacuum_boot () != NO_ERROR || prm_get_bool_value (PRM_ID_DISABLE_VACUUM) == true */
+  if (!vacuum_is_safe_to_remove_archives ())
+    {
+      return;
+    }
+
+  min_log_pageid = vacuum_min_log_pageid_to_keep (thread_p);
+  if (min_log_pageid == NULL_PAGEID)
+    {
+      /* this is an assertion case but ignore. */
+      return;
+    }
+
+  fprintf (outfp, "\n");
+  fprintf (outfp, "*** Vacuum Dump ***\n");
+  fprintf (outfp, "First log page ID referenced = %lld ", min_log_pageid);
+
+  if (logpb_is_page_in_archive (min_log_pageid))
+    {
+      LOG_CS_ENTER (thread_p);
```

Comment:
I think LOG_CS_ENTER() is not necessary for here. 
because, there's no modification on log_Gl or log buffer, 
and LOG_ARCHIVE_CS_ENTER() is already used in logpb_is_page_in_archive(). 
how about using LOG_CS_ENTER_READ_MODE(thread_p) instead of LOG_CS_ENTER()?

Reply:
@JoohoK You might've mean logpb_get_archive_number(), not logpb_is_page_in_archive().

Reply:
Yes, Sorry, you are right. LOG_ARCHIVE_CS_ENTER() is in logpb_get_archive_number().

Reply:
There is 'assert (LOG_CS_OWN (thread_p))' in the logpb_fetch_from_archive() function. The call graph of this function is '... -> logpb_get_archive_number() -> logpb_fetch_from_archive()'. Can I replace LOG_CS_ENTER() to LOG_CS_ENTER_READ_MODE() as you mentioned?

Reply:
Oh, sorry. I read the comments incorrectly, and I left the answers incorrectly. I read LOG_CS_ENTER_READ_MODE() incorrectly with LOG_ARCHIVE_CS_ENTER() wrote in vice versa. Let's replace it with LOG_CS_ENTER_READ_MODE().

Reply:
OK, it sounds good